### DISCUSSION TITLE
[Bug Enhancement] Adding sub-items

### DISCUSSION THREAD

**Discussion by rogercreagh** [TechbeeAT/jtxBoard#717] [Ideas]
The current way of selecting an existing sub-item to link to a parent is not smooth.

There is also some confusion between quickly adding a new subnote/subtask and selecting an existing item as subnote/subtask.

1. The box at the top of the screen with `Add a subtask` which can be changed to `Add a subnote` using the three dots menu is only for quickly adding a new item it seems. It might be better if the text said `Quick add new subnote` or `Quick add new subtask` tomake it clearer that this was just for adding a new item without leaving the existing item screen.
2. Currently the top box always defaults to `Add a subtask` which is a bit annoying if you seldom use tasks, but do use notes. It would be better perhaps if it remembered the last thing you did. Or if it was a preset option which would be the default.
3. To add an existing item as a sub item (task or note) you have to go to the edit screen and if you are not displaying the `Feedback/Linked Notes` (or `Subtasks`) element on the default screen (because you don't often use it and you want to minimise clutter) then you have to select show all options and find the element.  It would be nice if the `Link subitem` options were shown in the three dots menu on the view item screen together with `Quick add new sub note/task`. Or even better simply add the link item icon at the right hand side of the top box so you would have both add new and link existing available at a single click.
4. The way the selection dialogue to link an existing item works is cumbersome. Firstly it shows nothing but the empty search box, only once you type a single character does the list appear. If you  then delete the character,or type enter then the list appears. Couldn't the full list be there by default?
5. The list includes journal, notes AND tasks. The same interface is used for both subtasks and feedback/linked notes. If I have selected Link a subtask then I expect only tasks to be shown. Or only notes for subnotes. If you want to select a journal entry as a sub item then there should be a third option to select a journal entry for a subitem. 
You could still use the same interface by adding a (possibly hidden) filter to it to show only one type of item.
6. The selection list includes items from ALL collections - this may be technically allowed but seems wrong - normally I would not want to cross link between vcalendars. Again this could be a filter option on the list - all collections or current item collection.
7. The list of items is sorted by date order ascending. Generally you are likely to be linking to a relatively recent item as a new subitem so perhaps sorting by date descending would be better (or again an option)
8. Because the list shows both summary (title) and an extract of the description each item takes up quite a lot of space so you only get a few on screen at a time. Personally I would prefer to have just title and categories listed. 
9. Which leads to the potential confusion as to what is being searched - currently it is both title and description, I would find it more useful to also be able to search by category. How about prefixing the search text with `d:` to search only in description, `c:` to search only in category, `s:` to search only in summary, and no prefix to search everything.

More than enough to chew on I think - perhaps you would like me to break this into 9 separate issues?

**Comment by patrickunterwegs** [comment]
Hi @rogercreagh , I'm moving this to discussions to pick out specific changes as an issue.

**Comment by patrickunterwegs** [comment]
> 2. Currently the top box always defaults to Add a subtask which is a bit annoying if you seldom use tasks, but do use notes. It would be better perhaps if it remembered the last thing you did. Or if it was a preset option which would be the default.

This is actually a bug and will be solved in #718

**Comment by patrickunterwegs** [comment]
> The box at the top of the screen with Add a subtask which can be changed to Add a subnote using the three dots menu is only for quickly adding a new item it seems. It might be better if the text said Quick add new subnote or Quick add new subtask tomake it clearer that this was just for adding a new item without leaving the existing item screen.

It's no problem to change this. The problem here is more about translations. While it's still short in English, translations for "Quick add new subnote" might get very long (e.g. in German) - too long for the field. But still, I can change it for English. 
By the way feel free to make any proposals to the english version, which should also be possible on the translation platform Crodwin (see the website for the link).

**Reply by patrickunterwegs** [reply]
Ok, sorry, I was a bit too fast with the answer, I reused another string there. I will add a new string with the more fitting text.

**Comment by patrickunterwegs** [comment]
> The list includes journal, notes AND tasks. The same interface is used for both subtasks and feedback/linked notes. If I have selected Link a subtask then I expect only tasks to be shown. Or only notes for subnotes. If you want to select a journal entry as a sub item then there should be a third option to select a journal entry for a subitem.
You could still use the same interface by adding a (possibly hidden) filter to it to show only one type of item.
The selection list includes items from ALL collections - this may be technically allowed but seems wrong - normally I would not want to cross link between vcalendars. Again this could be a filter option on the list - all collections or current item collection.
The list of items is sorted by date order ascending. Generally you are likely to be linking to a relatively recent item as a new subitem so perhaps sorting by date descending would be better (or again an option)

I will consider these points in #729

**Comment by patrickunterwegs** [comment]
About points 8 and 9 I am not in favor of such an implementation and I will leave them up for discussion.

**Reply by rogercreagh** [reply]
Yes, ignore 8 as I was forgetting Compact view which is mostly what I want. I'll add a separate discussion about a super-compact list option with no description.

Likewise for 9 I agree what I suggested initially is clunky (I was thinking of a different desktop system where using a short-code prefix to limit searching to a specific field is used, but I've not seen it implemented anywhere else and would be potentially confusing on android. I see if I can come upwith a better suggestion.
